factory/apiResolverFactory.go

```diff
@@ -331,6 +336,8 @@ func createScQueryElement(
 		BlockChainHook:    vmFactory.BlockChainHookImpl(),
 		BlockChain:        args.dataComponents.Blockchain(),
 		ArwenChangeLocker: args.coreComponents.ArwenChangeLocker(),
+		Accounts:          args.stateComponents.AccountsAdapter(),
```

Comment:
warning: this will provide the writable accounts adapter pointer.
Should use `args.stateComponents.AccountsAdapterAPI()`

Reply:
no need after removing it from the struct